docs/website/contents/for-developers/Whitepaper.md

```diff
@@ -0,0 +1,244 @@
+
+## Mempool & TxSubmission
+
+The Mempool maintains a sequence of transactions that could inhabit a hypothetical block that is valid and extends the current local selection.
+The Mempool is bounded via a multi-dimensional notion of size such that it never contains more transactions than could fit in N blocks.
+Peers synchronize their Mempools via the TxSubmission protocol.
+This mini protocol leverages the fact that the Mempool is a sequence as opposed to an unordered set; a simple integer suffices as the iterator state for the TxSub client.
+(Recall that transactions flow from client to server, since the orientation is determined by the flow of blocks and pending transactions naturally flow opposite of blocks.)
+
+## The Mint aka Block Forge
+
+Whenever the wall clock enters a new slot, the node checks whether its configured minting credentials (if any) were elected by the protocol state and forecasted ledger state to lead this slot.
+If so, it mints a block that extends its local selection (or its immediate predecessor if the tip somehow already occupies this slot or greater) and contains the longest prefix of the Mempool's transactions that can fit.
+That block is then sent to ChainSel, as if it had been fetched.
+
+When (if) the node selects that block, the Mempool will react as it does for any change in the local selection: it discards any transactions that are no longer valid in the updated hypothetical block the Mempool continuously represents.
+Because every Ouroboros transaction _consumes_ at least one UTxO, the transactions in a newly minted and selected block will definitely be discarded.
+
+## Some Important Optimizations
+
+Both both blocks and transactions can be applied much more quickly if they are known to have been previously validated.
+For blocks, the outcome will be exact same, since they can only validly extend a single chain.
+A transaction, though, might be validated against a different chain/ledger state than the first time, so many checks still need to happen.
+But many "static" checks don't need to be repeated, since they'd fail regardless of the ledger state.
+
+Despite block reapplication being much faster than initial validation, the node should not need to reapply their entire historical chain whenever it is restarted.
+The node instead occasionally persists its oldest ledger state (ie the kcp+1st).
+On startup, the node only needs to deserialize that snapshotted ledger state and then replay the best chain amongs the persisted blocks it has that extends this ledger state in order to re-establish its kcp+1 ledger states.
+
+# Multi-era Considerations
+
+In Cardano, an _era_ is an informal concept that unites a set of Cardano features, such as ledger rules or consensus protocols. In Cardano, the on-chain governance ultimately decides when to adopt backwards-incompatible changes, by incrementing the major component of the protocol version. The first such _era transition_ switched to the Praos protocol.
```

Comment:
I think it might be worth linking to this https://github.com/cardano-foundation/CIPs/blob/master/CIP-0059/feature-table.md